pkg/apis/nfd/nodefeaturerule/rule.go

```diff
@@ -31,32 +31,56 @@ import (
 	"sigs.k8s.io/node-feature-discovery/pkg/utils"
 )
 
+type RunStrategy interface {
+	ContinueExec() bool
+}
+
+type RunAllStrategy struct{}
+
+func (r RunAllStrategy) ContinueExec() bool {
+	return true
+}
+
+type FailFastStrategy struct{}
+
+func (f FailFastStrategy) ContinueExec() bool {
+	return false
+}
+
 // RuleOutput contains the output out rule execution.
 // +k8s:deepcopy-gen=false
 type RuleOutput struct {
+	Name              string
 	ExtendedResources map[string]string
 	Labels            map[string]string
 	Annotations       map[string]string
 	Vars              map[string]string
 	Taints            []corev1.Taint
+	Matches           MatchedFeatures
```

Comment:
@marquiz, what do you think about transforming the `MatchedFeatures` type into a type that reports not only features that match with the node but also those that don’t, perhaps called `CheckedFeatures`? This way, I wouldn’t need to introduce [complex comparison functions](https://github.com/kubernetes-sigs/node-feature-discovery/blob/10972517a430ca4dee4fc06aa31e3a1a7d599332/pkg/client-nfd/compat/validate-node.go#L61-L64) to identify which expressions passed or didn’t. For image compatibility, it’s also important to return information about expressions that don’t match the node.

In several places, there are log messages with matched features, which we can retain by adding a filter for the `CheckedFeatures` type. What do you think?

Reply:
After trying it out, it turned out to be too invasive and computation time to be longer for each rule processing. I've decided to simplify implementation on the cli side.